pandas/tests/series/test_constructors.py

```diff
@@ -393,6 +395,32 @@ def test_constructor_default_index(self):
         s = Series([0, 1, 2])
         tm.assert_index_equal(s.index, pd.Index(np.arange(3)))
 
+    @pytest.mark.parametrize('input', [[1, 2, 3],
+                                       (1, 2, 3),
+                                       list(range(3)),
+                                       pd.Categorical(['a', 'b', 'a']),
+                                       (i for i in range(3)),
+                                       map(lambda x: x, range(3))])
+    def test_constructor_index_mismatch(self, input):
+        # GH 19342
+        # test that construction of a Series with an index of different length
+        # raises an error
+        idx = np.arange(4)
+        if compat.PY2:
+            typs = types.GeneratorType
```

Comment:
you don't need all of this, its just confusing just to construct an error message. just do a simpler check on the error.